geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -9990,6 +9996,26 @@ public EntryExpiryTask getEntryExpiryTask(Object key) {
     return br.getEntryExpiryTask(key);
   }
 
+  @Override
+  public void updatePRNodeInformation(CacheLoader cacheLoader, CacheWriter cacheWriter) {
+    Region<String, PartitionRegionConfig> partionedRegionRoot = getPRRoot();
+    if (partionedRegionRoot != null) {
+      PartitionRegionConfig prConfig = partionedRegionRoot.get(getRegionIdentifier());
+      if (prConfig != null) {
+        for (Node node : prConfig.getNodes()) {
+          if (node.getMemberId().equals(getMyId())) {
+            byte loaderByte = (byte) (cacheLoader != null ? 0x01 : 0x00);
+            byte writerByte = (byte) (cacheWriter != null ? 0x02 : 0x00);
+            prConfig.removeNode(node);
```

Comment:
this "removeNode" call scares me. It seems like it could cause intermittent concurrency issues.
Until "prConfig.addNode" is called this node is no longer in the list. Note that other threads in this same jvm would be getting the same prConfig instance from the prRoot region. Other members will not see our changes until we do the put. It might be better to create a copy of the PartitionedRegionConfig (see getPRConfigWithLatestExpirationAttributes for an example) but you will need to copy the nodes.

Reply:
Done